src/industry_cmd.cpp

```diff
@@ -1709,16 +1712,16 @@ static void PopulateStationsNearby(Industry *ind)
 
 /**
  * Put an industry on the map.
- * @param i       Just allocated poolitem, mostly empty.
- * @param tile    North tile of the industry.
- * @param type    Type of the industry.
- * @param it      Industrylayout to build.
- * @param layout  Number of the layout.
- * @param t       Nearest town.
- * @param founder Founder of the industry; OWNER_NONE in case of random construction.
+ * @param i                   Just allocated poolitem, mostly empty.
+ * @param tile                North tile of the industry.
+ * @param type                Type of the industry.
+ * @param layout              Industrylayout to build.
+ * @param layout_index        Number of the industry layout.
+ * @param t                   Nearest town.
+ * @param founder             Founder of the industry; OWNER_NONE in case of random construction.
  * @param initial_random_bits Random bits for the industry.
  */
-static void DoCreateNewIndustry(Industry *i, TileIndex tile, IndustryType type, const IndustryTileTable *it, byte layout, Town *t, Owner founder, uint16 initial_random_bits)
+static void DoCreateNewIndustry(Industry *i, TileIndex tile, IndustryType type, const IndustryTileLayout &layout, byte layout_index, Town *t, Owner founder, uint16 initial_random_bits)
```

Comment:
not "your bug", but this function uses byte for layout_index/itspec_index, whereas `CheckIfIndustryTilesAreFree` uses uint. Should be consistent